src/models/plot.ts

```diff
@@ -23,13 +24,26 @@ export interface PlotObject {
   spec: FacetedCompositeUnitSpec;
 }
 
-export function extractPlotObjects(modelGroup: SpecQueryGroup<PlotObject>) {
+export function extractPlotObjects(modelGroup: SpecQueryGroup<PlotObject>,
+                                   filters: Array<RangeFilter|OneOfFilter>): PlotObject[] {
   return modelGroup.items.map(item => {
     if (isSpecQueryGroup<PlotObject>(item)) {
       const childModelGroup = item as SpecQueryGroup<PlotObject>;
-      return getTopSpecQueryItem(childModelGroup);
+      return {
+        ...getTopSpecQueryItem(childModelGroup),
```

Comment:
this seems redundant to call twice?

Let's make a method that takes a spec and filters as input can combine them and re-use it in both cases.